src/Webapi/Dom/Webapi__Dom__HtmlTextAreaElement.res

```diff
@@ -0,0 +1,170 @@
+
+  @get external autocomplete: t_htmlTextAreaElement => string = "autocomplete"
+  @set external setAutocomplete: (t_htmlTextAreaElement, string) => unit = "autocomplete"
+
+  @get external cols: t_htmlTextAreaElement => float = "cols"
+  @set external setCols: (t_htmlTextAreaElement, float) => unit = "cols"
+
+  @get external dirName: t_htmlTextAreaElement => string = "dirName"
+  @set external setDirName: (t_htmlTextAreaElement, string) => unit = "dirName"
+
+  @get external disabled: t_htmlTextAreaElement => bool = "disabled"
+  @set external setDisabled: (t_htmlTextAreaElement, bool) => unit = "disabled"
+
+  @get external maxLength: t_htmlTextAreaElement => float = "maxLength"
+  @set external setMaxLength: (t_htmlTextAreaElement, float) => unit = "maxLength"
+
+  @get external minLength: t_htmlTextAreaElement => float = "minLength"
+  @set external setMinLength: (t_htmlTextAreaElement, float) => unit = "minLength"
+
+  @get external placeholder: t_htmlTextAreaElement => string = "placeholder"
+  @set external setPlaceholder: (t_htmlTextAreaElement, string) => unit = "placeholder"
+
+  @get external readOnly: t_htmlTextAreaElement => bool = "readOnly"
+  @set external setReadOnly: (t_htmlTextAreaElement, bool) => unit = "readOnly"
+
+  @get external required: t_htmlTextAreaElement => bool = "required"
+  @set external setRequired: (t_htmlTextAreaElement, bool) => unit = "required"
+
+  @get external rows: t_htmlTextAreaElement => float = "rows"
+  @set external setRows: (t_htmlTextAreaElement, float) => unit = "rows"
+
+  @get external wrap: t_htmlTextAreaElement => string = "wrap"
+  @set external setWrap: (t_htmlTextAreaElement, string) => unit = "wrap"
+
+  @get external textLength: t_htmlTextAreaElement => float = "textLength"
+
+  /* Validation */
+  @get external willValidate: t_htmlTextAreaElement => bool = "willValidate"
+  @get external validity: t_htmlTextAreaElement => Webapi__Dom__ValidityState.t = "validity"
+  @get external validationMessage: t_htmlTextAreaElement => string = "validationMessage"
+  @send external setCustomValidity: (t_htmlTextAreaElement, string) => unit = "setCustomValidity"
+  @send external checkValidity: t_htmlTextAreaElement => bool = "checkValidity"
+  @send external reportValidity: t_htmlTextAreaElement => bool = "reportValidity"
+
+  /* Selection */
+  @send external select: t_htmlTextAreaElement => unit = "select"
+  @get external selectionStart: t_htmlTextAreaElement => int = "selectionStart"
+  @set external setSelectionStart: (t_htmlTextAreaElement, int) => unit = "selectionStart"
+  @get external selectionEnd: t_htmlTextAreaElement => int = "selectionEnd"
+  @set external setSelectionEnd: (t_htmlTextAreaElement, int) => unit = "selectionEnd"
+  @get external selectionDirection: t_htmlTextAreaElement => string = "selectionDirection"
+  @set
+  external setSelectionDirection: (t_htmlTextAreaElement, string) => unit = "selectionDirection"
+
+  module SelectionDirection = {
+    type t =
```

Comment:
Why can't these be polymorphic?

Reply:
These were copied from `Webapi__Dom__HtmlInputElement.res`, so it's just what was there already. I'd prefer not to duplicate the code in both places but wasn't sure what you all thought before posting.

Reply:
Our long term aim is to reduce the runtime code - when this library was written polymorphic variants were not strings. We can't convert existing modules until we do a wider breaking API change release in 2022, but for new modules we can make them as good as ReScript bindings allow.

So yes, please convert to polymorphic variants without runtime wrapping code.